### PR TITLE
Flatten Swift telemetry dictionary

### PR DESCRIPTION
**Pull request by thomasameisel**
Flatten the Swift telemetry dictionary in the telemetryDictionary function as opposed to within the host app

### CODE REVIEW

File: compiler/cpp/src/thrift/generate/t_java_generator.cc [hunk 1]
```diff
@@ -5292,7 +5292,6 @@ void t_java_generator::generate_java_struct_tuple_writer(ofstream& out, t_struct
     }
 
     indent(out) << "oprot.writeBitSet(optionals, " << optional_count << ");" << endl;
-    int j = 0;
```
Comment (GyroJoe) [inline compiler/cpp/src/thrift/generate/t_java_generator.cc:5295]: This was just unused?

File: compiler/cpp/src/thrift/generate/t_swift_generator.cc [hunk 2]
```diff
@@ -970,26 +938,57 @@ void t_swift_generator::telemetry_dictionary_value(ofstream& out, t_type* type,
...
-
-    block_close(out, false);
-    out << "())";
   } else if (type->is_enum()) {
+    out << indent() << "telemetryData[\"" << name << "\"] = ";
     out << ".string(" << property_name << ".telemetryName(), piiKind: " << pii_kind << ")";
+    out << endl;
   } else if (type->is_struct()) {
-    out << ".dictionary(" << property_name << ".telemetryDictionary())";
+    out << indent() << "telemetryData.merge(" << property_name << ".telemetryDictionary()) { _, child in child }";
```
Comment (GyroJoe) [inline compiler/cpp/src/thrift/generate/t_swift_generator.cc:957]: Do you think we should be warning consumers in the case of a conflict?

**Review by GyroJoe — APPROVED** [review]

File: compiler/cpp/src/thrift/generate/t_java_generator.cc (hunk 1 above)
Comment (thomasameisel) [reply compiler/cpp/src/thrift/generate/t_java_generator.cc:5295]: Yep it was giving me an error when compiling locally

File: compiler/cpp/src/thrift/generate/t_swift_generator.cc (hunk 2 above)
Comment (thomasameisel) [reply compiler/cpp/src/thrift/generate/t_swift_generator.cc:957]: We have a lint rule [here](https://github.com/outlook/thrift/blob/ms-master/compiler/cpp/src/thrift/lint/t_linter.cc#L296) to check for members in sub-structs and maps with enum keys. We're not able to check at compile-time for maps with string keys. Do you have ideas for how we would warn consumers? FWIW we didn't have any logs for this even in Outlook

**Pull request merged**